tic-tac-toe/src/app/components/game/index.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+
+import Board from '../board';
+
+import style from './styles.scss';
+
+class Game extends Component {
+  state = {
+    history: [{ squares: Array(9).fill(null) }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  calculateWinner = squares => {
+    const lines = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
+
+    for (let i = 0; i < lines.length; i += 1) {
+      const [a, b, c] = lines[i];
+      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+        return squares[a];
+      }
+    }
+    return null;
+  };
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (this.calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{ squares }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    });
+  };
+
+  jumpTo = step => {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  };
+
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = this.calculateWinner(current.squares);
+    const moves = history.map((step, move) => {
+      const desc = move ? `Go to move #${move}` : 'Go to game start';
+      return (
+        <li key={move}>
+          <button onClick={() => this.jumpTo(move)}>{desc}</button>
```

Comment:
extract the onClick function. you should always keep render components and JS code separately. Also, you have a problem here, you are using a function and you are passing a paramether, so you are telling the render to re render everytime it change... that's not cool because that function should not render everytime. how to solved it? the paramether `move` should be coming as prop. You should extract this render as a 'dummy view' (we will talk this) and pass everything as prop. Let's check this together tomorrow